bazel/glog.bzl

```diff
@@ -38,58 +78,65 @@ def glog_library(namespace = "google", with_gflags = 1, **kwargs):
             "src/utilities.cc",
             "src/utilities.h",
             "src/vlog_is_on.cc",
+        ] + select({
+            "@bazel_tools//src/conditions:windows": ["src/windows/port.cc", "src/windows/port.h"],
+            "//conditions:default": [":config_h"],
+        }),
+        copts =
+            select({
+                "@bazel_tools//src/conditions:windows": common_copts + windows_only_copts,
+                "@bazel_tools//src/conditions:darwin": common_copts + linux_or_darwin_copts + darwin_only_copts,
+                "//conditions:default": common_copts + linux_or_darwin_copts,
+            }),
+        deps = [
+            ":glog_headers",
+        ] + gflags_deps,
+        **kwargs
+    )
+
+    # glog headers vary depending on the os.
+    native.cc_library(
+        name = "glog_headers",
+        deps = select({
+            "@bazel_tools//src/conditions:windows": [":windows_glog_headers"],
+            "//conditions:default": [":default_glog_headers"],
+        }),
+    )
+
+    native.cc_library(
+        name = "windows_glog_headers",
+        hdrs = ["src/glog/log_severity.h", "src/windows/config.h",] + native.glob(["src/windows/glog/*.h"]),
+        includes = ["src/windows"],
+        # config.h for windows seem hardcoded that way,
+        # and we need to propagate those defines to binaries/libraries linking
+        # against glog.
+        defines = [
+            "GOOGLE_GLOG_IS_A_DLL=1",
+            "GOOGLE_GLOG_DLL_DECL=__declspec(dllexport)",
+            "GOOGLE_GLOG_DLL_DECL_FOR_UNITTEST=__declspec(dllimport)",
```

Comment:
Are these necessary in `defines`? If so, are they all necessary or can some be in `copts`?

It looks to me like the public headers have an `#ifndef GOOGLE_GLOG_DLL_DECL` section that should make them unnecessary, but perhaps I'm missing something.